tests/test_hydra.py

```diff
@@ -1399,3 +1400,45 @@ def test_frozen_primary_config(
     cmd.extend(overrides)
     ret, _err = run_python_script(cmd)
     assert expected in ret
+
+
+@mark.parametrize(
+    "env_deprecation_err,expected",
+    [
+        param(
+            False,
+            r"^\S*/my_app\.py:9: UserWarning: Feature FooBar is deprecated$",
+            id="deprecation_warning",
+        ),
+        param(
+            True,
+            dedent(
+                r"""
+                ^Error executing job with overrides: \[\]\n?
+                Traceback \(most recent call last\):
+                  File "\S*/my_app.py", line 9, in my_app
+                    hydra\.\S*\.deprecation_warning\("Feature FooBar is deprecated"\)
+                  File "\S*\.py", line 9, in deprecation_warning
+                    raise DeprecationWarning\(.*\)
+                DeprecationWarning: Feature FooBar is deprecated
+
+                Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace\.$
+                """
+            ).strip(),
+            id="deprecation_error",
+        ),
+    ],
+)
+def test_hydra_deprecation_warning(
+    monkeypatch: Any, env_deprecation_err: bool, expected: str, tmpdir: Path
+) -> None:
+    cmd = [
+        "tests/test_apps/deprecation_warning/my_app.py",
+        f"hydra.run.dir={tmpdir}",
+    ]
+    if env_deprecation_err:
+        monkeypatch.setenv("HYDRA_DEPRECATION_WARNINGS_AS_ERRORS", "1")
+    _, err = run_python_script(
+        cmd, allow_warnings=True, print_error=False, raise_exception=False
+    )
```

Comment:
There is a run_with_error function that returns stderr even if there is an exception.
Can you try to use it instead of changing run_python_script?

Reply:
The `run_with_error` function adds the `-W` flag to the command, which makes warnings get turned into errors. This makes it harder to tell whether `HYDRA_DEPRECATION_WARNINGS_AS_ERRORS` is set or not.

Reply:
run_python_script is also preexisting, right?

I believe you can pass an env dictionary to run_python_script instead of mocking the environment like this.

Reply:
> run_python_script is also preexisting, right?

Yes.

> I believe you can pass an env dictionary to run_python_script instead of mocking the environment like this.

You're right. Made the update in https://github.com/facebookresearch/hydra/pull/1703/commits/2c4d9513018ec0e722cf21c626f0c2a5f65065d1.